kube/Cargo.toml

```diff
@@ -16,8 +16,8 @@ edition = "2018"
 
 [features]
 default = ["native-tls"]
-native-tls = ["openssl", "reqwest/native-tls"]
-rustls-tls = ["rustls", "reqwest/rustls-tls"]
+native-tls = ["openssl", "reqwest/native-tls", "real-native-tls", "tokio-native-tls", "async-tungstenite/tokio-native-tls"]
+rustls-tls = ["rustls", "reqwest/rustls-tls", "tokio-rustls", "async-tungstenite/tokio-rustls"]
```

Comment:
you have done really good feature work here

Reply:
i would probably add another feature for websockets, since using these is kind of a niche feature, at least for the more controller heavy crowd that we are going for - however that will probably make the feature juggling you are doing here more difficult

+ websockets = ["async-tungstenite", ..]

Reply:
Yeah, definitely. I honestly don't have any plans to use it at the moment. It was just technically interesting to fit this into `kube` :D

Maybe something like:
```toml
[features]
default = ["native-tls"]
native-tls = ["openssl", "reqwest/native-tls"]
rustls-tls = ["rustls", "reqwest/rustls-tls"]
derive = ["kube-derive"]
ws = ["async-tungstenite"]
ws-native-tls = ["tokio-native-tls", "async-tungstenite/tokio-native-tls"]
ws-rustls-tls = ["tokio-rustls", "async-tungstenite/tokio-rustls"]
```

For code, splitting into separate modules should help a lot.

Reply:
Hah, well it's very welcome. It's needed to get to client-gold. Btw, with all these contributions, feel free to pop your name on the authors list in kube and kube-derive's `Cargo.toml`s - I am happy to give you some more push access :-)

Feature wise, yeah, that looks sensible. Picking ws + a tls stack for it. Can't think of a way to get around two features to pick for this.

Reply:
Hm, features aren't allowed to enable features of a disabled dependency?

Reply:
I remember seeing an RFC for that. I'll look into it.

If it's allowed, we can do:
```toml
[features]
default = ["native-tls"]
native-tls = ["openssl", "reqwest/native-tls", "async-tungstenite/tokio-native-tls"]
rustls-tls = ["rustls", "reqwest/rustls-tls", "async-tungstenite/tokio-rustls"]
derive = ["kube-derive"]
ws = ["async-tungstenite"]
```

Reply:
Currently, it'll always enable the dependency itself. Tracking issue: https://github.com/rust-lang/cargo/issues/8832